src/formatDate.js

```diff
@@ -50,7 +50,62 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const newDate = date.split(fromFormat[3]);
+  const toSeparate = toFormat[3];
+  const resultDate = [];
+  let month = '';
+  let day = '';
+  let year = '';
+
+  for (let i = 0; i < newDate.length; i++) {
+    switch (fromFormat[i]) {
+      case 'MM':
+        month = newDate[i];
```

Comment:
```suggestion
        dateObject.MM = newDate[i];
```
Use masks as keys. You need to have all possible values: 'DD', 'MM', 'YY' and 'YYYY'